server/main.go

```
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+)
+
+type product struct {
+	ID          int    `json:"id"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Price       int    `json:"price"`
+}
+
+var products []product
+
+func getProductsHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(products); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+func returnProductHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id, err := strconv.ParseInt(vars["id"], 10, 64)
+	if err != nil {
+		http.Error(w, "Unable to convert string to int", http.StatusInternalServerError)
+	}
+
+	for _, product := range products {
+		if int64(product.ID) == id {
+			if err = json.NewEncoder(w).Encode(product); err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+			}
+		}
+	}
+}
+
+func createNewProductHandler(w http.ResponseWriter, r *http.Request) {
+	reqBody, _ := ioutil.ReadAll(r.Body)
```

You're discarding this error. Handle it.